src/fsharp/FSharp.Compiler.Service.ProjectCracker/ProjectCracker.fs

```diff
@@ -20,7 +20,7 @@ type ProjectCracker =
             let referencedProjects = Array.map (fun (a, b) -> a, convert b) opts.ReferencedProjectOptions
             
             let sourceFiles, otherOptions = 
-                opts.Options |> Array.partition (fun x -> Path.GetExtension(x).ToLower() = ".fs")
+                opts.Options |> Array.partition (fun x -> x.IndexOfAny(Path.GetInvalidPathChars()) = -1 && Path.GetExtension(x).ToLower() = ".fs")
```

Comment:
Ah, that looks cleaner but that part of the code base seems quite independent. Thanks for the fast review.